File: Crio/crio_programming_interview_problems-master/FindPeakElement/FindPeakElement.java
import java.io.*;
import java.util.*;

public class FindPeakElement {
    // Implement your solution by completing the below function
    public int findPeakElement(int[] nums) {
        // CRIO_SOLUTION_START_MODULE_L2_PROBLEMS
        int left = 0, right = nums.length - 1;
        while (left < right) {
            int mid = left + ((right - left) / 2);
            if (nums[mid] > nums[mid + 1])
                right = mid;
            else
                left = mid + 1;
        }
        return left;
        // CRIO_SOLUTION_END_MODULE_L2_PROBLEMS

        // CRIO_UNCOMMENT_START_MODULE_L2_PROBLEMS
        //return 0;
        // CRIO_UNCOMMENT_END_MODULE_L2_PROBLEMS
    }

    public static void main(String[] args) throws IOException {
        Scanner scanner = new Scanner(System.in);

        int[] nums = new int[scanner.nextInt()];
        for (int i = 0; i < nums.length; i++)
            nums[i] = scanner.nextInt();
        scanner.close();

        int peakElement = new FindPeakElement().findPeakElement(nums);
        System.out.print(peakElement);
    }
}

File: Crio/crio_programming_interview_problems-master/FindPeakElement/FindPeakElement.cpp
#include "FastIO.hpp"
#include "ReadMatrix.hpp"
#include <bits/stdc++.h>
using namespace std;

class FindPeakElement {
  public:
    int findPeakElement(vector<int>& nums) {
        // TODO: CRIO_TASK_MODULE_L2_PROBLEMS
        // Your implementation goes here
        // CRIO_SOLUTION_START_MODULE_L2_PROBLEMS
        int l = 0, r = nums.size() - 1;
        while (l < r) {
            int mid = (l + r) / 2;
            if (nums[mid] > nums[mid + 1])
                r = mid;
            else
                l = mid + 1;
        }
        return l;
        // CRIO_SOLUTION_END_MODULE_L2_PROBLEMS
    }
};

int main() {
    FastIO();
    int n;
    cin >> n;
    vector<int> nums;
    ReadMatrix<int>().OneDMatrix(n, nums);
    int result = FindPeakElement().findPeakElement(nums);
    cout << result;
    return 0;
}